### PR TITLE
Add "search for lyrics in google" link if lyrics could not be found 

### PR DESCRIPTION
**Pull request by dKab**
Closes (#2675)
image

### CODE REVIEW

File: src/renderer/ui/components/generic/LyricsViewer.js [hunk 1]
```diff
@@ -1,21 +1,31 @@
 import $ from 'jquery';
 import React, { Component, PropTypes } from 'react';
 import { findDOMNode } from 'react-dom';
-
+import { shell, remote } from 'electron';
 import { requireSettings } from './SettingsProvider';
 
-class LyricsViewer extends Component {
+export class LyricsViewerComponent extends Component {
```
Comment (dKab) [inline src/renderer/ui/components/generic/LyricsViewer.js]: The default export is remaining untouched, so the modules that are using LyricsViewer won't break. This additional export of the component itself (not a HOC wrapped around it) is required in order to be able to call its method in test suite. To call class methods of component rendered with enzyme we need to call `component.instance()` to get the instance of its class. If we import the HOC exported by default, the call to `component.instance()` returns wrong object which doesn't have methods we want to test.

Comment (MarshallOfSound) [reply src/renderer/ui/components/generic/LyricsViewer.js]: I'd rather we don't test a component that isn't technically being rendered directly.  You can fetch the inner component to call it's method using the enzyme `find` method.

https://github.com/airbnb/enzyme/blob/master/docs/api/ShallowWrapper/find.md

File: test/electron-renderer/ui/LyricsViewer_spec.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
-/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-expressions, no-duplicate-imports, import/no-duplicates */
```
Comment (dKab) [inline test/electron-renderer/ui/LyricsViewer_spec.js]: some of the tests are actually expecting HOC wrapper which is being exported by default and won't pass if we use pure component class. The new tests require the unwrapped component itself in order to call its methods directly. So, I had to turn off this rule for this file and import two things from one file.  The other option is not to turn off eslint but instead create separate file for HOC wrapper.

Comment (dKab) [inline test/electron-renderer/ui/LyricsViewer_spec.js]: Some of the tests are actually expecting HOC wrapper which is being exported by default and won't pass if we use pure component class. The new tests require the unwrapped component itself in order to call its methods directly. So, I had to turn off this rule for this file and import two things from one file.  The other option is not to turn off eslint but instead create separate file for HOC wrapper.

File: src/renderer/ui/components/generic/LyricsViewer.js (hunk 1 above)
Comment (dKab) [reply src/renderer/ui/components/generic/LyricsViewer.js]: Got it, will try `find`

**Comment by dKab** [comment]
@MarshallOfSound the solution with `find` method that you suggested didn't work for some reason:
image

But, I've found a way to test this indirectly (which is even better, to my mind). I just wait till `handleLyricsNotFound()` will be called automatically when 4 seconds have passed.  Sinon has a handy method for emulating passing of time which allows tests to be fast. I also changed one existing test to use this technique. Now tests run time has dropped by 4.5 seconds and we don't need to export underlying component after all.

**Comment by MarshallOfSound** [comment]
> Now tests run time has dropped by 4.5 seconds and we don't need to export underlying component after all.

Nice!!! 👍

**Comment by MarshallOfSound** [comment]
Thanks for this 😄

**Comment by sayoojsamuel** [comment]
The `Search on google.com` option often searches with the query 
```
*<song name> - Unknown Artist lyrics*
```
This often result in a Google search without any Rich Text lyrics panel.  In the case of unknown artist, just a *<song name> lyrics* query would suffice, with better chance for formatted lyrics.

**Pull request merged**